common/queue/src/main/java/org/thingsboard/server/queue/provider/KafkaTbCoreQueueFactory.java

```diff
@@ -150,6 +154,7 @@ public KafkaTbCoreQueueFactory(PartitionService partitionService, TbKafkaSetting
         consumerBuilder.groupId("tb-core-node");
         consumerBuilder.decoder(msg -> new TbProtoQueueMsg<>(msg.getKey(), ToCoreMsg.parseFrom(msg.getData()), msg.getHeaders()));
         consumerBuilder.admin(coreAdmin);
+        consumerBuilder.statsService(consumerStatsService);
```

Comment:
Since this is a component and you already autowire the consumerStatsService, you can just register the group id inside this Factory and avoid changes to the TbKafkaConsumerTemplate.

Reply:
but this way I won't be able to 'unregister' group on 'unsubscribe'
it will be useful if we have some logic of deleting queues (and therefore consumers)